src/main/scala/com/mongodb/spark/config/ReadConfig.scala

```diff
@@ -122,6 +129,27 @@ object ReadConfig extends MongoInputConfig {
     notNull("options", options)
     apply(sparkConf, options.asScala)
   }
+
+  private def getPartitioner(partitionerName: String): MongoPartitioner = {
+    val partitionerClassName = if (partitionerName.contains(".")) partitionerName else s"$DefaultPartitionerPath$partitionerName"
+    val parsedPartitioner = Try({
+      val clazz = Class.forName(partitionerClassName)
+      partitionerClassName.endsWith("$") match {
+        case true  => clazz.getField("MODULE$").get(clazz).asInstanceOf[MongoPartitioner]
```

Comment:
Is there a test for it?  I see that HalfwayPartitioner is now a class rather than an object.